ra2ce/analysis/losses/losses_base.py

```diff
@@ -464,23 +506,38 @@ def _calculate_production_loss_per_capita(
         disrupted is then obtained multiplying the traffic intensity by the total occupancy per vehicle type,
         including drivers, by the daily loss of productivity per capita per hour.
 
-        the unit of time is hour.
+        The unit of time is hour.
         """
         vlh_total = 0
+        _relevant_link_type = self._get_relevant_link_type(vlh_row, row_hazard_range)
+        _divisor = self._get_divisor(_relevant_link_type, row_hazard_range)
+
+        duration_steps = self.resilience_curve.get_duration_steps(
+            _relevant_link_type, row_hazard_range
+        )
+        functionality_loss_ratios = self.resilience_curve.get_functionality_loss_ratio(
+            _relevant_link_type, row_hazard_range
+        )
+
         for trip_type in self.trip_purposes:
             intensity_trip_type = self.vot_intensity_per_trip_collection[
                 f"intensity_{self.part_of_day}_{trip_type}"
             ].loc[[vlh_row[self.link_id]]]
             occupancy_trip_type = float(
                 self.vot_intensity_per_trip_collection[f"occupants_{trip_type}"]
             )
-            # TODO: improve formula based on road_type, water_heigth resilience_curve.csv (duration*loss_ratio) instead of duration_event
-            # Compare with other function
-            vlh_trip_type_event_series = (
-                self.duration_event
-                * intensity_trip_type
-                * occupancy_trip_type
-                * self.production_loss_per_capita_per_hour
+            vlh_trip_type_event_series = sum(
+                (
+                    intensity_trip_type
+                    * duration
+                    * loss_ratio
```

Comment:
There is no loss_ratio to be considered here. 
The fornula is: Intensity * duration *occupancy * production_loss

Reply:
Except this point, all good to me :)